### ISSUE TITLE
do correct label transformation when specifying DataEnvironment

### ISSUE THREAD

**Issue by cosmicBboy** [cosmicBboy/kiosk-ml#72]
there's a bug where training fails when target classes are strings (not already numericalized)

https://github.com/cosmicBboy/kiosk-ml/blob/master/kiosk/models/training_env.py#L31-L34

use target_preprocessor argument when initializing the `DataEnvironment`

https://github.com/cosmicBboy/kiosk-ml/blob/master/kiosk/models/training_env.py#L37

So that string target arrays are converted to [numerical codes](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.LabelEncoder.html)
